fineract-provider/src/main/java/org/apache/fineract/useradministration/starter/UserAdministrationAutoConfiguration.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.fineract.useradministration.service.AppUserWritePlatformServiceJpaRepositoryImpl;
+import org.apache.fineract.useradministration.service.PasswordPreferencesWritePlatformServiceJpaRepositoryImpl;
+import org.apache.fineract.useradministration.service.PasswordValidationPolicyReadPlatformServiceImpl;
+import org.apache.fineract.useradministration.service.PermissionReadPlatformServiceImpl;
+import org.apache.fineract.useradministration.service.PermissionWritePlatformServiceJpaRepositoryImpl;
+import org.apache.fineract.useradministration.service.RoleDataValidator;
+import org.apache.fineract.useradministration.service.RoleReadPlatformService;
+import org.apache.fineract.useradministration.service.RoleReadPlatformServiceImpl;
+import org.apache.fineract.useradministration.service.RoleWritePlatformServiceJpaRepositoryImpl;
+import org.apache.fineract.useradministration.service.UserDataValidator;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+@Configuration
+public class UserAdministrationAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
+    public AppUserReadPlatformServiceImpl appUserReadPlatformService(PlatformSecurityContext context, JdbcTemplate jdbcTemplate,
+            OfficeReadPlatformService officeReadPlatformService, RoleReadPlatformService roleReadPlatformService,
+            AppUserRepository appUserRepository, StaffReadPlatformService staffReadPlatformService) {
+        return new AppUserReadPlatformServiceImpl(context, jdbcTemplate, officeReadPlatformService, roleReadPlatformService,
+                appUserRepository, staffReadPlatformService);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public AppUserWritePlatformServiceJpaRepositoryImpl appUserWritePlatformServiceJpaRepository(PlatformSecurityContext context,
+            UserDomainService userDomainService, PlatformPasswordEncoder platformPasswordEncoder, AppUserRepository appUserRepository,
+            OfficeRepositoryWrapper officeRepositoryWrapper, RoleRepository roleRepository, UserDataValidator fromApiJsonDeserializer,
+            AppUserPreviousPasswordRepository appUserPreviewPasswordRepository, StaffRepositoryWrapper staffRepositoryWrapper,
+            ClientRepositoryWrapper clientRepositoryWrapper) {
+        return new AppUserWritePlatformServiceJpaRepositoryImpl(context, userDomainService, platformPasswordEncoder, appUserRepository,
+                officeRepositoryWrapper, roleRepository, fromApiJsonDeserializer, appUserPreviewPasswordRepository, staffRepositoryWrapper,
+                clientRepositoryWrapper);
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public PasswordPreferencesWritePlatformServiceJpaRepositoryImpl passwordPreferencesWritePlatformServiceJpaRepository(
```

Comment:
Don't use implementation classes here, always use Java interfaces... they are not excheangable! Please provide the Java interface class reference in parenthesis with the `@ConditionalOnMissingBean` annotation.

Reply:
Done.